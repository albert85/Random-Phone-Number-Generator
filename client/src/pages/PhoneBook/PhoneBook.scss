@import "../../assets/variable";

.phone-book {
  display: grid;
  grid-gap: 30px;
  padding: 0 150px;
  height: 100vh;
  font-family: sans-serif;
  font-weight: 100;
  background-color: $background;
  justify-content: space-between;
  grid-template-columns: repeat(3, 1fr);
  grid-row-gap: 30px;
  grid-template-rows: 10% 27% 1fr;
  grid-template-areas:
    "header header header"
    "total-report-card max-report-card min-report-card"
    "side-menu table-view table-view";

  .card {
    background-color: $white;
    padding: 20px;
    box-shadow: $boxshadow;
  }

  .text-size {
    font-size: 20px;
  }

  .text-value-size {
    text-align: right;
    margin-top: 57px;
    font-size: 20px;
  }

  .header {
    grid-area: header;
    padding-top: 30px;
    font-family: sans-serif;
    font-size: 30px;
  }

  .total-phone-number {
    grid-area: total-report-card;
  }

  .max-phone-number {
    grid-area: max-report-card;
  }

  .min-phone-number {
    grid-area: min-report-card;
  }

  .side-menu {
    grid-area: side-menu;
    background-color: $white;
    box-shadow: $boxshadow;
    height: 580px;
    padding: 30px;

    & > p {
      margin: 0;
      font-size: 18px;
      font-family: sans-serif;
      font-weight: 100;
      margin-bottom: 10px;
    }
  }

  .table-view {
    grid-area: table-view;
    height: 600px;
    overflow-y: auto;
    padding: 20px;
    background-color: $white;
    box-shadow: $boxshadow;
  }

  .number-list {
    display: flex;
    flex-wrap: wrap;
    list-style-type: none;
    margin: 0;
    padding: 0;
    overflow-y: auto;
    justify-content: center;

    & > li {
      margin: 10px;
      background-color: $white;
      padding: 17px;
      width: 126px;
      border: 1px solid map-get($palette, primary);
      text-align: center;
    }
  }

  .sort-container {
    & > p {
      margin: 0 0 10px 0;
    }

    .sort-field {
      width: 100%;
      height: 60px;
      color: blue;
      border: 1px solid;
      background-color: white;
      font-size: 17px;
    }
  }

  .generate-container {
    & > p {
      margin: 20px 0 10px 0;
    }

    & > span {
      color: map-get($palette, danger);
      position: absolute;
      margin-top: -22px;
    }
  }

  .adjust-size {
    width: 100%;
    height: 60px;
    font-size: 18px;
    margin-bottom: 20px;
    border: 1px solid map-get($palette, primary);

    &--generate,
    &--get-all {
      height: 60px;
      width: 100%;
      margin-bottom: 20px;
      box-shadow: $boxshadow;
      font-size: 18px;
      border: 1px solid map-get($palette, primary);
      color: map-get($palette, primary);
    }

    &--generate:hover,
    &--get-all:hover {
      color: $white;
      cursor: pointer;
      background-color: map-get($palette, primary);
    }
  }
}
