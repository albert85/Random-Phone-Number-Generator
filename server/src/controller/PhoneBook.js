import { checkLimit, checkSort } from 'helper/validator';
import { GenerateAndSave } from 'helper/util';

export default class PhoneBook {
  static generateTelephoneNo(req, res) {
    const { limit, sort } = req.query;

    const AmountOfTelNoGenerated = checkLimit(limit);
    const unsortedResult = GenerateAndSave(AmountOfTelNoGenerated);
    const sortedTelephoneNumbers = checkSort(sort, unsortedResult);

    return res.status(201).json({ message: 'Telephone successfully generated', data: sortedTelephoneNumbers });
  }
}
