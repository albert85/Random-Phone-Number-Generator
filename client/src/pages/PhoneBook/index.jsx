import React from 'react';

import './PhoneBook.scss';

export default class PhoneBook extends React.Component {
  renderCard(cardName, titleHeaderTitle, cardValue) {
    return (
      <div className={`card ${cardName}`}>
        <p className="text-size">{titleHeaderTitle}</p>
        <p className="text-value-size">{cardValue}</p>
      </div>
    )
  }

  render() {
    return (
      <div className="phone-book">
        <div className="header"></div>
        {this.renderCard('total-phone-number', 'Total Generated Number', 0)}
        {this.renderCard('max-phone-number', 'Maximum Number Generated', 5000)}
        {this.renderCard('min-phone-number', 'Minimum Number Generated', 10000)}
        <div className="side-menu">
          <h1>Menu</h1>
        </div>
        <div className="table-view">
          <h1>table</h1>
        </div>
      </div>
    )
  }
}
