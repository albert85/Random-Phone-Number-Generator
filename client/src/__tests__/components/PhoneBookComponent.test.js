import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import moxios from 'moxios';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, mount } from 'enzyme';

import { PhoneBook } from '../../pages/PhoneBook';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

// Initialize mockstore with empty state
const initialState = {};
const store = mockStore(initialState);

Enzyme.configure({ adapter: new Adapter() });

describe('Random Number', () => {
    const response = {
        message: "",
        max: "",
        min: "",
        data: []
    }
    const props = {
        generatePhoneNumbers: jest.fn(),
        getAllPhoneNumbers: () => Promise.resolve({
            message: "Telephone successfully generated",
            max: "0808331345",
            min: "0154646646",
            data: ["0808331345", "0154646646", "0254646646"]
        }),
        phoneBook: {
            ...response
        }
    }

    const component = shallow(<PhoneBook {...props} />)

    beforeEach(() => {
        moxios.install();
    });

    afterEach(() => {
        moxios.uninstall();
    });

    it('should render the page', () => {
        expect(component).toMatchSnapshot();
    });

    it('should get all generated numbers', () => {
        moxios.wait(() => {
            const request = moxios.requests.mostRecent();
            request.respondWith({
              status: 201,
              response: {
                ...response
              }
            });
          });
          const component = mount(<PhoneBook {...props} />)
        // console.log('before', component.instance().state);
        component.find('.get-all-btn').simulate('click');
        // console.log('after', component.instance().state);
        //   console.log(component.debug());
    });

})
