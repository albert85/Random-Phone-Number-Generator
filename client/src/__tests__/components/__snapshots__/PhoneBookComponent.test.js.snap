// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random Number should render the page 1`] = `ShallowWrapper {}`;
